langchain/src/vectorstores/typesense.ts

```diff
@@ -0,0 +1,370 @@
+      pageContent: text,
+      metadata: metadatas[i] || {},
+    }));
+    await instance.addDocuments(documents);
+
+    return instance;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async addVectors(_vectors: number[][]) {
+    throw new Error("Method not implemented");
+  }
+
+  /**
+   * Create a retriever from the vector store.
+   * @returns Typesense retriever
+   */
+  asRetriever(
+    k?: number,
+    filter?: this["FilterType"]
+  ): VectorStoreRetriever<this> {
+    return new VectorStoreRetriever<this>({
+      vectorStore: this,
+      filter,
+      k,
+    });
+  }
+
+  /**
+   * Transform the Typesense records to documents.
+   * @param typesenseRecords
+   * @returns documents
+   */
+  typesenseRecordsToDocuments(
+    typesenseRecords: Record<string, any>[] | undefined
+  ): Document[] {
+    const documents =
+      typesenseRecords?.map((hit) => {
+        const objectWithMetadatas: Record<string, any> = {};
+
+        this.metadataColumnNames.forEach((metadataColumnName) => {
+          objectWithMetadatas[metadataColumnName] = hit[metadataColumnName];
+        });
+
+        const document: Document = {
+          pageContent: hit[this.pageContentColumnName] || "",
+          metadata: objectWithMetadatas,
+        };
+        return document;
+      }) || [];
+
+    return documents;
+  }
+
+  /**
+   * Search for similar documents.
+   * @param query query
+   * @param k amount of results to return
+   * @param filter filter to apply to the search, merged with default search params. As Langchain chains method does do the similarity search implicitly, you can use modifySearchParams to change the default search params.
+   * @returns similar documents
+   */
+  async similaritySearch(
+    query: string,
+    k?: number,
+    filter: Record<string, any> = {}
+  ) {
+    const amount = k || this.searchParams.per_page || 5;
+    const vectorPrompt = await this.embeddings.embedQuery(query);
+    const vector_query = `${this.vectorColumnName}:([${vectorPrompt}], k:${amount})`;
+    const typesenseResponse = await this.client.multiSearch.perform(
+      {
+        searches: [
+          {
+            ...this.searchParams,
+            ...filter,
+            per_page: amount,
+            vector_query,
+            collection: this.schemaName,
+          },
+        ],
+      },
+      {}
+    );
+
+    const results = typesenseResponse.results[0].hits;
+    const hits = results?.map((hit) => hit.document) as
+      | Record<string, any>[]
+      | undefined;
+
+    const documents = this.typesenseRecordsToDocuments(hits);
+
+    return documents;
+  }
+
+  /**
+   * Delete documents from the vector store.
+   * @param documentIds ids of the documents to delete
+   */
+  async deleteDocuments(documentIds: string[]) {
+    await this.client
+      .collections(this.schemaName)
+      .documents()
+      .delete({
+        filter_by: `id:=${documentIds.join(",")}`,
+      });
+  }
+
```

Comment:
I believe both `similaritySearchVectorWithScore` and `similaritySearchWithScore` can be used in other chains / retrievers. afaict the TypeSense API does not provide the similarity scores in their APIs, thus would it make sense to just use `1` as a score instead?

Reply:
Has sense to me, I added the methods